cmp/filter/FilterFieldModel.js

```diff
@@ -0,0 +1,296 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2020 Extremely Heavy Industries Inc.
+ */
+
+import {XH, HoistModel, managed, PersistenceProvider} from '@xh/hoist/core';
+import {fmtNumber} from '@xh/hoist/format';
+import {action, observable} from '@xh/hoist/mobx';
+import {Filter, FilterModel, parseFieldValue} from '@xh/hoist/data';
+import {throwIf} from '@xh/hoist/utils/js';
+import {differenceWith, isEmpty, isEqual, isNil, isPlainObject, groupBy, sortBy, map, take, partition} from 'lodash';
+
+import {FilterOptionsModel} from './FilterOptionsModel';
+
+@HoistModel
+export class FilterFieldModel {
+
+    @observable.ref value;
+    @observable.ref history;
+
+    // Immutable properties
+    filterModel = null;
+    filterOptionsModel = null;
+    limit = null;
+    maxHistoryLength = null;
+
+    @managed provider;
+
+    /**
+     * @param c - FilterFieldModel configuration.
+     * @param {(FilterModel|Object)} c.filterModel - FilterModel, or config to create one.
+     * @param {(FilterOptionsModel|Object)} c.filterOptionsModel - FilterOptionsModel, or config to create one.
+     * @param {number} [c.limit] - maximum number of results to show before truncating.
+     * @param {PersistOptions} [c.persistWith] - options governing history persistence
+     * @param {number} [c.maxHistoryLength] - number of recent selections to maintain in the user's
+     *      history (maintained automatically by the control on a FIFO basis).
+     */
+    constructor({
+        filterModel,
+        filterOptionsModel,
+        limit,
+        persistWith,
+        maxHistoryLength = 5
+    }) {
+        throwIf(!filterModel, 'Must provide a FilterModel (or a config to create one)');
+        throwIf(!filterOptionsModel, 'Must provide a FilterOptionsModel (or a config to create one)');
+
+        if (isPlainObject(filterModel)) {
+            this.filterModel = this.markManaged(new FilterModel(filterModel));
+        } else {
+            this.filterModel = filterModel;
+        }
+
+        if (isPlainObject(filterOptionsModel)) {
+            this.filterOptionsModel = this.markManaged(new FilterOptionsModel(filterOptionsModel));
+        } else {
+            this.filterOptionsModel = filterOptionsModel;
+        }
+
+        this.limit = limit;
+        this.maxHistoryLength = maxHistoryLength;
+
+        // Read state from provider -- fail gently
+        if (persistWith) {
+            try {
+                this.provider = PersistenceProvider.create({path: 'filterField', ...persistWith});
+                const state = this.provider.read();
+                if (state?.history) this.history = state.history;
```

Comment:
should we store/rehydrate value as well?

Reply:
Honestly, the persistence isn't really in place yet... that's on the list of things to discuss